### PR TITLE
feat: GAP-3 & GAP-4 (model evaluation metrics and export)

### PR DESCRIPTION
**Pull request by Sickwoman**
## Summary
Fixes GAP-3 and GAP-4 from the backend gaps analysis in a single PR.

---

### Problem
After training, only a single loss/metric number was shown:
```python
loss, metric = model.evaluate(x_testing, y_testing)
_model_result(f"Final Metric: {metric}", 3)
# That's it. One number. No per-class breakdown, no confusion matrix, nothing.
```
A binary classifier could get 95% accuracy while completely failing on the minority class — and the user would have no idea.

### Changes
- **New model**: `ModelEvaluation` table stores detailed metrics per training run
- **New model**: `ModelTrainingRun` table (self-contained on this branch, tracks every run)
- **New service**: `app/services/evaluation.py` computes:
  - Per-class precision, recall, F1-score (classification)
  - Confusion matrix (classification)
  - ROC AUC score + full curve data (binary classification only)
  - MAE, RMSE, R² score (regression)
- **Updated**: `model_run.py` now captures `history = model.fit(...)` and saves full evaluation after training

### New Endpoint
| Method | Endpoint | Description |
|--------|----------|-------------|
| `GET` | `/api/v1/model/{model_id}/evaluation` | Get detailed evaluation metrics, optionally filtered by `?run_id=` |

---

### Problem
- Model trained and evaluated — weights immediately discarded
- No way to download or reuse the trained model outside TensorMap

### Changes
- **New service**: `app/services/model_export.py` handles exporting trained weights
- Supports two formats:
  - `h5` — Keras HDF5 format, single file download
  - `pb` — TensorFlow SavedModel format, zipped directory

### New Endpoint
| Method | Endpoint | Description |
|--------|----------|-------------|
| `GET` | `/api/v1/model/{model_id}/export?format=h5` | Download model as `.h5` file |
| `GET` | `/api/v1/model/{model_id}/export?format=pb` | Download model as `.zip` SavedModel |

---

## Other Changes
- Added `scikit-learn` dependency for classification metrics
- Alembic migration `6cbbfc386658` creates `model_training_run` and `model_evaluation` tables
- `migrations/env.py` updated to include new models for autogenerate

## No Breaking Changes
All existing endpoints and behaviour are completely unchanged.

### CODE REVIEW

**Comment by ivantha** [comment]
Closing this in line with the TensorMap `gsoc` label convention. Issue #120 reserves the full scope of the GSoC 2026 student's term, and this PR overlaps with **Phase 4 (Model export + comparison)** and **Phase 5 (Model interpretability and post-training analysis: confusion matrix, classification report, feature importance)** of that scope.

Coding for the selected contributor (Kartikey Gupta) begins on 2026-05-25, so these surfaces are fenced off until then. Thanks for the contribution; if you would like to contribute before or around GSoC, please pick up a `pre-gsoc` labeled issue instead.
